Fixed the issue when --password was used without a value

bsd-conqueror opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Improvement

Short description (up to few sentences):

Fixed the issue when the option "--password" was used without a value, i.e.
`
clickhouse client --host 127.0.0.1 --password
`
In this case, trigger the "--ask-password" processing to prompt for a password from the user.

Detailed description (optional):
1. Fixed the issue when a blank user is shown in the password prompt

Commit:
Fixed the issue when --password was used without a value

alexey-milovidov commented on dbms/programs/client/ConnectionParameters.h:
```diff
@@ -67,6 +66,18 @@ struct ConnectionParameters
         else
         {
             password = config.getString("password", "");
+            if (password == "")
```
config.has

alexey-milovidov commented on dbms/programs/client/ConnectionParameters.h:
```diff
@@ -67,6 +66,18 @@ struct ConnectionParameters
+            if (password == "")
+            {
+                // std::cout << "--password was used but set to empty string, switching to password prompt.";
+                std::cout << "Password for user " << user << ": ";
```
Better to avoid code duplication.

Commit:
Fixed the issues with the --password option and the user name in the prompt

- if --password was used without any value, ask for the password in the prompt
- fixed the issue when a blank user name is shown in the prompt if no user was provided

bsd-conqueror committed:
Merge branch 'master' into missing_password_value

Commit:
Removed code duplication in --password handling

Commit:
Merge branch 'missing_password_value' of https://github.com/bsd-conqueror/ClickHouse into missing_password_value

Commit:
Added a comment to explain why '\n' is used as the implicit value for the --password option

Commit:
Fixed a mistype

alexey-milovidov committed:
Update Client.cpp

alexey-milovidov committed:
Update ConnectionParameters.h

filimonov commented on dbms/programs/client/Client.cpp:
```diff
@@ -1547,7 +1547,13 @@ class Client : public Poco::Util::Application
+              * the "\n" is used to distinguish this case because there is hardly a chance an user would use "\n"
+              * as the password.
+              */
+            ("password", po::value<std::string>()->implicit_value("\n"), "password")
```
that leads to bad-looking --help:
```clickhouse-client --help
Main options:
  --help                                                           produce help message
  -C [ --config-file ] arg                                         config-file path
  -c [ --config ] arg                                              config-file path (another shorthand)
  -h [ --host ] arg (=localhost)                                   server host
  --port arg (=9000)                                               server port
  -s [ --secure ]                                                  Use TLS connection
  -u [ --user ] arg (=default)                                     user
  --password [=arg(=
)]                                            password
  --ask-password                                                   ask-password
```
